From 835b1980fa31fabb64142b60d0a900d3118bae5a Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Mon, 11 Sep 2017 14:10:19 -0700
Subject: [PATCH] Don't attempt to parse EXIF data for raw files

This is causing crashes since raw files can have strange EXIF data.

Change-Id: Ie2bb2b3d27ec484b14c9dfa7a8f31ed9ea31d5c2
---
 media/java/android/media/MediaScanner.java   | 3 +--
 media/java/android/media/ThumbnailUtils.java | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/media/java/android/media/MediaScanner.java b/media/java/android/media/MediaScanner.java
index f565af0525d..e46f53b370a 100644
--- a/media/java/android/media/MediaScanner.java
+++ b/media/java/android/media/MediaScanner.java
@@ -946,8 +946,7 @@ public class MediaScanner implements AutoCloseable {
                 values.put(Audio.Media.IS_ALARM, alarms);
                 values.put(Audio.Media.IS_MUSIC, music);
                 values.put(Audio.Media.IS_PODCAST, podcasts);
-            } else if ((mFileType == MediaFile.FILE_TYPE_JPEG
-                    || MediaFile.isRawImageFileType(mFileType)) && !mNoMedia) {
+            } else if ((mFileType == MediaFile.FILE_TYPE_JPEG) && !mNoMedia) {
                 ExifInterface exif = null;
                 try {
                     exif = new ExifInterface(entry.mPath);
diff --git a/media/java/android/media/ThumbnailUtils.java b/media/java/android/media/ThumbnailUtils.java
index abd6f4a49ac..fa2f9db7ad5 100644
--- a/media/java/android/media/ThumbnailUtils.java
+++ b/media/java/android/media/ThumbnailUtils.java
@@ -92,8 +92,7 @@ public class ThumbnailUtils {
         SizedThumbnailBitmap sizedThumbnailBitmap = new SizedThumbnailBitmap();
         Bitmap bitmap = null;
         MediaFileType fileType = MediaFile.getFileType(filePath);
-        if (fileType != null && (fileType.fileType == MediaFile.FILE_TYPE_JPEG
-                || MediaFile.isRawImageFileType(fileType.fileType))) {
+        if (fileType != null && (fileType.fileType == MediaFile.FILE_TYPE_JPEG)) {
             createThumbnailFromEXIF(filePath, targetSize, maxPixels, sizedThumbnailBitmap);
             bitmap = sizedThumbnailBitmap.mBitmap;
         }
-- 
2.11.0

